lib/quicly.c

```diff
@@ -3337,18 +3339,24 @@ static int handle_reset_stream_frame(quicly_conn_t *conn, quicly_reset_stream_fr
     return 0;
 }
 
-static int handle_ack_frame(quicly_conn_t *conn, size_t epoch, quicly_ack_frame_t *frame)
+static int handle_ack_frame(quicly_conn_t *conn, struct st_quicly_handle_payload_state_t *state)
 {
+    quicly_ack_frame_t frame;
     quicly_sentmap_iter_t iter;
-    uint64_t packet_number = frame->smallest_acknowledged;
     struct {
         uint64_t packet_number;
         int64_t sent_at;
     } largest_newly_acked = {UINT64_MAX, INT64_MAX};
     size_t segs_acked = 0, bytes_acked = 0;
     int includes_ack_eliciting = 0, ret;
 
-    switch (epoch) {
+    ++state->num_frames_non_ack_eliciting;
```

Comment:
where is this being used?

Reply:
I see it now.

Reply:
It's being used in handle_payload function to determine if the packet was ack-eliciting.

Though I think it might make more sense to add a column to the handler table (defined in the `handle_payload` function) that designates if the handler is handling an ack-eliciting frame. Then, the determination (of if the payload of a packet is ack eliciting) can be done solely in `handle_payload`.